tests/ui/job-view/Filtering_test.jsx

```diff
@@ -0,0 +1,275 @@
+      fetchMock.get(
+        getProjectUrl('/bug-job-map/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/bug_suggestions/', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/text_log_steps/', 'autoland'),
+        [],
+      );
+    });
+
+    const setFilterText = (filterField, text) => {
+      fireEvent.click(filterField);
+      fireEvent.change(filterField, { target: { value: text } });
+      fireEvent.keyDown(filterField, { key: 'Enter', code: 13 });
+    };
+
+    test('string "yaml" should have 10 jobs', async () => {
+      const { getAllByText, findAllByText } = render(<App />);
+      await waitForElement(() => findAllByText('B'));
+      const filterField = document.querySelector('#quick-filter');
+      setFilterText(filterField, 'yaml');
+
+      await waitForElementToBeRemoved(() => getAllByText('B'));
+      expect(jobCount()).toBe(10);
+
+      // undo the filtering and make sure we see all the jobs again
+      setFilterText(filterField, null);
+      await waitForElement(() => getAllByText('B'));
+      expect(jobCount()).toBe(40);
+    });
+
+    test('click signature should have 10 jobs', async () => {
+      const { getByText, getByTitle, findAllByText } = render(<App />);
+      const build = await waitForElement(() => findAllByText('B'));
+
+      fireEvent.mouseDown(build[0]);
+      const sigLink = await waitForElement(() => getByText('(sig)'));
+      expect(sigLink.getAttribute('href')).toBe(
+        '/#/jobs?repo=autoland&selectedJob=259537372&searchStr=2aa083621bb989d6acf1151667288d5fe9616178',
+      );
+
+      const keywordLink = await waitForElement(() =>
+        getByTitle('Filter jobs containing these keywords'),
+      );
+      expect(keywordLink.getAttribute('href')).toBe(
+        '/#/jobs?repo=autoland&selectedJob=259537372&searchStr=gecko%2Cdecision%2Ctask%2Copt%2Cgecko%2Cdecision%2Ctask%2C%28d%29',
+      );
+    });
+  });
+
+  describe('by result status', () => {
+    const clickFilterChicklet = color => {
+      fireEvent.click(document.querySelector(`.btn-${color}-filter-chicklet`));
+    };
+
+    test('uncheck success should leave 30 jobs', async () => {
+      const { getAllByText, findAllByText } = render(<App />);
+      await waitForElement(() => findAllByText('B'));
+      clickFilterChicklet('green');
+
+      await waitForElementToBeRemoved(() => getAllByText('D'));
+      expect(jobCount()).toBe(30);
+
+      // undo the filtering and make sure we see all the jobs again
+      clickFilterChicklet('green');
+      await waitForElement(() => getAllByText('D'));
+      expect(jobCount()).toBe(40);
+    });
+
+    test('uncheck failures should leave 20 jobs', async () => {
+      const { getAllByText, findAllByText } = render(<App />);
+      const symbolToRemove = 'B';
+      await waitForElement(() => findAllByText('B'));
```

Comment:
You can do `findAllByText(symbolToRemove)`

Reply:
Oh shoot.  Good catch!  :)